UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/MetaDataLayer.cs

```diff
@@ -473,4 +473,81 @@ public void UpdateSystemsAt(Vector3Int localPosition, SystemType ToUpDate = Syst
 	{
 		subsystemManager.UpdateAt(localPosition, ToUpDate);
 	}
+
+	public void StoreReagentsAtTile(ReagentMix reagents, Vector3Int localPosInt)
+	{
+		var cellReagents = HasReagentsAtTile(localPosInt);
+		ReagentMix excess = null;
+		if (cellReagents.Item1)
+		{
+			excess = cellReagents.Item2.Split(reagents.Total / 2);
+			cellReagents.Item2.Add(reagents);
+		}
+		else
+		{
+			tileReagentMixes.Add(localPosInt, reagents);
+			if (reagents.Total <= REAGENT_LIMIT_PER_CELL)
+			{
+				excess = cellReagents.Item2.Split(REAGENT_LIMIT_PER_CELL);
+			}
+			CreateLiquidOverlay(localPosInt, reagents);
+		}
+		if (excess != null) DistributeExcessToNearbyCells(excess, localPosInt);
+	}
+
+	public Tuple<bool, ReagentMix> HasReagentsAtTile(Vector3Int localPosInt)
+	{
+		if (tileReagentMixes.TryGetValue(localPosInt, out var mix))
+		{
+			return Tuple.Create(true, mix);
+		}
+		return Tuple.Create(false, new ReagentMix());
+	}
+
+	private void DistributeExcessToNearbyCells(ReagentMix excess, Vector3Int origin)
```

Comment:
Does this mean if you drop a humongous amount will instantly fill the room?

Reply:
Yes and no.
Yes, it will do that if a puddle is already filled.
No, it will need at least one puddle for the distribution to occur. If there are no puddles, it's only going to be one tile for the time being. This behavior is intended to stop people from creating semen stations more easily.